### PR TITLE
Setup Experience: Empty string path should be treated as invalid path

### PR DESCRIPTION
**Pull request by gl-srgr**
TLDR: we have several scenarios (described in more detail in another comment) where `root` property of the Other local repo configuration is empty string, usually when we want to skip discovery/sync of local repos altogether. `filepath.Abs("")` call means we'll end up discovering on current working directory. This is a swift fix and we can revisit empty string + local sync'd repos behavior after Starship.

---- 

In setup wizard we display currently sync'd local repos on step 1:
image

This is done by calling [`localDiscovery()`](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/cmd/frontend/graphqlbackend/app.go?L85) graphql query and using the path obtained from [`localExternalServices() `](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/cmd/frontend/graphqlbackend/app.go?L147). 

In this previous [PR]( https://github.com/sourcegraph/sourcegraph/pull/49744) we skip discovery in `Walk()` when path is empty string, and in this new PR we add a similar check for the graphql discovery call. Without this new check we'll report repos in cwd even if we're not sync'ing them. E.g. in the above screenshot my service has `root:""` but still says we've got repos sync'ing. With the fix I get:

image

In this new check I return an error which is not displaying in the setup wizard but can be viewed in api console at least:
image

Predecessor / Related PR: https://github.com/sourcegraph/sourcegraph/pull/49744

## Test plan

sg start app SRC=""
run setup wizard -> check step 1 for currently sync'd local repos

### CODE REVIEW

**Comment by gl-srgr** [comment]
@vovakulikov  can you confirm for me that returning the error is not going to bubble up to the setup wizard or cause issues? As of now it appears that the error is ignored so it won't show up in ui but wanted to double check.

**Comment by gl-srgr** [comment]
I believe empty string is the common `root` value we have to handle as special case. I've got another ticket for when someone enters

`root: "/"`

and I don't believe we need to prioritize this for starship.

https://github.com/sourcegraph/sourcegraph/issues/49746

**Review by keegancsmith — APPROVED** [review]
I think it is correct to enforce this on the backend, but surely we should skip making the request at all in the frontend.

**Comment by gl-srgr** [comment]
@keegancsmith yeah agreed, just keeping frontend changes minimized for starship

**Comment by gl-srgr** [comment]
@vovakulikov Are you referring to this https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/client/web/src/setup-wizard/components/local-repositories-step/LocalRepositoriesStep.tsx?L177? Does this check for just nil and not empty string?

**Comment by gl-srgr** [comment]
I re ran my repro steps and it is mainly occurring with the autogenerated external service, so the reset path button isn't necessarily triggering the behavior but starting sg app with `SRC = ""` does trigger the behavior.

So I believe for the setup created service we check for empty string, but for autogenerated we do not check empty string. The repo discovered in or within current working directory only shows up in the Built-in repositories part of the wizard since that's referring to the autogenerated ext svc only. This new check prevents the discovery from returning the repos for empty string and so built-in repos will be empty.

**Comment by vovakulikov** [comment]
>Does this check for just nil and not an empty string?

First time I'm glad it works in this way in JavaScript; both empty string or null is false when it's casted to boolean which is not `!` operator does

**Comment by gl-srgr** [comment]
Gotcha, so I believe the empty string check is taking place for (2.) but not (1.) below, and I misunderstood when debugging:

```
1. Autogenerated service: pass SRC = "" env var

2. Setup wizard generated service: pick a path, revisit setup wizard later and click "reset path" -> updates existing external service to root:""
```

So this fix will still help in (1.) which is what I was seeing in my repro. It could be useful to do the check in frontend but for now the backend also removes the issue

**Comment by vovakulikov** [comment]
Oh, yeah you're right, we don't have empty directory path check for the autogenerated service, I will add this later this week
